internal/infrastructure/kubernetes/resource/resource.go

```diff
@@ -41,6 +44,24 @@ func CompareSvc(currentSvc, originalSvc *corev1.Service) bool {
 		cmpopts.IgnoreFields(corev1.ServiceSpec{}, "ClusterIP", "ClusterIPs"))
 }
 
+// CompareDeployment compare the current from the k8s and deployment from the resource_provider.
+func CompareDeployment(current, deployment *appv1.Deployment) bool {
+	// applied to k8s the "DeprecatedServiceAccount" will fill it.
+	deployment.Spec.Template.Spec.DeprecatedServiceAccount = current.Spec.Template.Spec.DeprecatedServiceAccount
```

Comment:
why not ignore this field using `cmp.Diff` ??

Reply:
too many fields need to ignore, and when new feature needs to be paid attention to here, and In order to maintain consistency with the following